substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/linux/LinuxSubstitutions.java

```diff
@@ -66,6 +72,46 @@ public static String mapLibraryName(String libname) {
     }
 }
 
+@Platforms(Platform.LINUX.class)
+@TargetClass(java.lang.Runtime.class)
+final class Target_java_lang_Runtime {
+
+    @Substitute
+    public int availableProcessors() {
+        if (SubstrateOptions.MultiThreaded.getValue()) {
+            LinuxSched.cpu_set_t cpuSet;
+            int size, res;
+            final int pid = Unistd.getpid();
```

Comment:
The manual page for `sched_getaffinity` says "If pid is zero, then the mask of the calling process is returned.", which would save you explicitly getting your pid.